pkg/reconciler/channel/channel.go

```diff
@@ -123,30 +124,42 @@ func (r *Reconciler) reconcile(ctx context.Context, c *v1alpha1.Channel) error {
 		return nil
 	}
 
-	channelResourceInterface := r.DynamicClientSet.Resource(duckroot.KindToResource(c.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())).Namespace(c.Namespace)
+	gvr, _ := meta.UnsafeGuessKindToResource(c.Spec.ChannelTemplate.GetObjectKind().GroupVersionKind())
+	channelResourceInterface := r.DynamicClientSet.Resource(gvr).Namespace(c.Namespace)
+	if channelResourceInterface == nil {
+		msg := fmt.Sprintf("Unable to create dynamic client for: %+v", c.Spec.ChannelTemplate)
+		logging.FromContext(ctx).Error(msg)
+		return errors.New(msg)
+	}
+
+	track := r.channelableTracker.TrackInNamespace(c)
 
-	// Tell tracker to reconcile this Channel whenever the backing Channel CRD changes.
-	track := r.resourceTracker.TrackInNamespace(c)
+	backingChannelObjRef := corev1.ObjectReference{
+		Kind:       c.Spec.ChannelTemplate.Kind,
+		APIVersion: c.Spec.ChannelTemplate.APIVersion,
+		Name:       c.Name,
+		Namespace:  c.Namespace,
+	}
+	// Tell the channelTracker to reconcile this Channel whenever the backing Channel changes.
+	if err := track(backingChannelObjRef); err != nil {
+		logging.FromContext(ctx).Error("Unable to track changes to the backing channel", zap.Error(err))
+		return err
```

Comment:
My preference is to simply annotate the original error and return it, as we know `Reconcile` will log it.

```suggestion
		return fmt.Errorf("track the channel: %v", err)
```

Reply:
Done, thanks!